Sources/SwiftParser/Statements.swift

```diff
@@ -1038,4 +1051,63 @@ extension Parser.Lookahead {
     } while lookahead.at(.poundIf, .poundElseif, .poundElse) && lookahead.hasProgressed(&loopProgress)
     return lookahead.atStartOfSwitchCase()
   }
+
+  /// Returns `true` if the current token represents the start of an `if` or `while` statement body.
+  mutating func atStartOfStatementBody() -> Bool {
```

Comment:
You're right. I'm bad with naming, your suggestion is more clear. I just tried replacing it by `atValidTrailingClosure` and most of my test cases failed. I don't think it worth trying to consolidating them, at least not now, because it's seems there are a few differences and I don't feel comfortable changing a function that's used in places outside the scope of my PR.

Reply:
Okay